### ISSUE TITLE
gateleen-delegate: include request headers in payload transformation

### ISSUE THREAD

**Issue by mcweba** [swisspush/gateleen#278]
The current implementation of the _Delegate_ class allows to transform the body of the delegated request only.
Sometimes it's necessary to also include request headers in the transformation.

See documentation of [Payload transformation](https://github.com/swisspush/gateleen/blob/develop/gateleen-delegate/README_delegate.md#payload-transformation)

#### How to realize
The input JSON which is sent to the _JoltTransformer_ has to be extended to include metadata (new) information and payload (already existing) information.

Example:
```json
{
	"metadata": {
		"key 1": "value 1",
		"key 2": "value 2",
		"key 3": "value 3"
	},
	"payload": {}
}
```
This solution should allow to let the _JoltTransformer_ as it is.

**Attention:** Existing transformation specs in delegates have to be updated since the payload is not the root node anymore!

**Comment by hiddenalpha** [comment]
Shouldn't we add another level as in:
```json
{
  "metadata": {
    "headers": {
      "key 1": "value 1",
      "key 2": "value 2"0
    }
  },
  "payload": {}
}
```
That would be more extensible. For example imagine we need to also include URL for some reason in that structure later. Without that additional nesting, we would be breaking again.

**Comment by lbovet** [comment]
No need fo the metadata level, then.

Input JSON for transform would be:
```
{
    "urlParts": string[],  // regexp pattern match groups on the URL
    "headers": {
      "Header-Key-1": [
        "value 1"
      ],
      "Header-Key-2": [
        "value 2a",
        "value 2b"
      ]
    },
   "payload": ..., // original request payload
}

**Comment by mcweba** [comment]
As discussed with @lbovet we introduce a new property in the delegate definition called _transformWithMetadata_ to not break jolt specifications on existing delegate definitions, since they depend on the payload to be the root node of the input json.

**Comment by mcweba** [comment]
Released in [v1.1.45](https://github.com/swisspush/gateleen/releases/tag/v1.1.45)
